monkey/common/agent_plugins/agent_plugin_manifest.py

```diff
@@ -31,3 +34,11 @@ class AgentPluginManifest(InfectionMonkeyBaseModel):
     description: Optional[str]
     link_to_documentation: Optional[str]
     safe: bool = False
+
+    @validator("name")
+    def validate_name(cls, name):
```

Comment:
This looks like a reusable filename validator. Maybe add a file `monkey/common/types/filename.py` and add a validator there (named `validate_filename`)?
Then in the `AgentPluginManifest` add
```
_validate_filename = validator('name', allow_reuse=True)(validate_filename)
```

Reply:
Seems unnecessary to me right now. We can move it when (if) we need to reuse it.